File: exercise/python/day03/preparation/SantaWorkshopService.py
from preparation.Gift import Gift


class SantaWorkshopService:
    def __init__(self):
        self.prepared_gifts = []

    def prepare_gift(self, gift_name: str, weight: float, color: str, material: str) -> Gift:
        if weight > 5:
            raise ValueError("Gift is too heavy for Santa's sleigh")

        gift = Gift(gift_name, weight, color, material)
        self.prepared_gifts.append(gift)
        return gift

File: exercise/C#/day03/Preparation/SantaWorkshopService.cs
namespace Preparation
{
    public class SantaWorkshopService
    {
        private readonly List<Gift> _preparedGifts = new();

        public Gift PrepareGift(string giftName, double weight, string color, string material)
        {
            if (weight > 5)
            {
                throw new ArgumentException("Gift is too heavy for Santa's sleigh");
            }

            var gift = new Gift(giftName, weight, color, material);
            _preparedGifts.Add(gift);

            return gift;
        }
    }
}